pyvo/mivot/viewer/mivot_viewer.py

```diff
@@ -0,0 +1,494 @@
+            The number corresponding to the resource containing the MIVOT block (first by default).
+        """
+        if not check_astropy_version():
+            raise AstropyVersionException(f"Astropy version {version.version} "
+                                          f"is below the required version 6.0 for the use of MIVOT.")
+        else:
+            if isinstance(votable_path, DALResults):
+                self._parsed_votable = votable_path.votable
+            elif isinstance(votable_path, VOTableFile):
+                self._parsed_votable = votable_path
+            else:
+                self._parsed_votable = parse(votable_path)
+            self._table_iterator = None
+            self._connected_table = None
+            self._connected_tableref = None
+            self._current_data_row = None
+            # when the search object is in GLOBALS
+            self._globals_instance = None
+            self._last_row = None
+            self._templates = None
+            self._resource = None
+            self._annotation_seeker = None
+            self._mapping_block = None
+            self._mapped_tables = []
+            self._resource_seeker = None
+            self._instance = None
+            try:
+                self._set_resource(resource_number)
+                self._set_mapping_block()
+                self._resource_seeker = ResourceSeeker(self._resource)
+                self._set_mapped_tables()
+                self._connect_table(tableref)
+                self._init_instance()
+            except MappingException as mnf:
+                logging.error(str(mnf))
+
+    """
+    Properties
+    """
+    def __enter__(self):
+        """ `with` statement implementation """
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """ `with` statement implementation """
+        logging.info("MivotViewer closing..")
+
+    def close(self):
+        """ `with` statement implementation """
+        logging.info("MivotViewer is closed")
+
+    @property
+    def votable(self):
+        """
+        returns the Astropy parsed votable
+        """
+        return self._parsed_votable
+
+    @property
+    def annotation_seeker(self):
+        """
+        Return an API to search various components in the XML mapping block.
+        """
+        return self._annotation_seeker
+
+    @property
+    def resource_seeker(self):
+        """
+        Return an API to search various components in the VOTabel resource.
+        """
+        return self._resource_seeker
+
+    @property
+    def connected_table(self):
+        """
+        getter for the identifier the astropy.table
+        instance the viewer is connected to
+        """
+        return self._connected_table
+
+    @property
+    def connected_table_ref(self):
+        """ getter for the identifier the table the viewer is connected to """
+        return self._connected_tableref
+
+    @property
+    def instance(self):
```

Comment:
Not sure if it's the best name but I don't have a better suggestion. `dm_object`? 

Or maybe use it as an iterator with `next()` returning the instance? But that is not doable for someone looking for the xml representation.

Reply:
`Mivot_Viewer.(_)instance` attribute replaced with `Mivot_Viewer.(_)dm_instance`